polyfill/lib/duration.mjs

```diff
@@ -279,69 +282,124 @@ export class Duration {
     const maximum = maximumIncrements[smallestUnit];
     if (maximum !== undefined) ES.ValidateTemporalRoundingIncrement(roundingIncrement, maximum, false);
 
+    let zonedRelativeTo = ES.IsTemporalZonedDateTime(relativeTo) ? relativeTo : undefined;
+    let plainRelativeTo = ES.IsTemporalDate(relativeTo) ? relativeTo : undefined;
```

Comment:
This was already merged in #2657 where it was changed somewhat:
```js
let { plainRelativeTo, zonedRelativeTo } = ES.ToRelativeTemporalObject(roundTo);
```
`plainRelativeTo` needs to be mutable because later we calculate it if `zonedRelativeTo` was provided but we determine we'll need the `plainRelativeTo` counterpart for an operation. `zonedRelativeTo` doesn't need to be mutable but I guess it's along for the ride, just for convenience in destructuring.